src/Sentry/SentryClient.cs

```diff
@@ -234,16 +234,18 @@ private bool CaptureEnvelope(Envelope envelope)
             catch (Exception e)
             {
                 _options.DiagnosticLogger?.LogError("The BeforeSend callback threw an exception. It will be added as breadcrumb and continue.", e);
-
-                @event?.AddBreadcrumb(
-                    "BeforeSend callback failed.",
-                    category: "SentryClient",
-                    data: new Dictionary<string, string>
-                    {
-                        {"message", e.Message},
-                        {"stackTrace", e.StackTrace}
-                    },
-                    level: BreadcrumbLevel.Error);
+                if (e.StackTrace != null)
+                {
+                    @event?.AddBreadcrumb(
+                        "BeforeSend callback failed.",
+                        category: "SentryClient",
+                        data: new Dictionary<string, string>
+                        {
+                            {"message", e.Message},
+                            {"stackTrace", e.StackTrace}
+                        },
+                        level: BreadcrumbLevel.Error);
+                }
```

Comment:
Why throw everything away if there's no stack trace?


```suggestion
                var data = new Dictionary<string, string>
                {
                    {"message", e.Message},
                };
                if (e.StackTrace is not null)
                {
                    data.Add("stackTrace", e.StackTrace);
                }
                @event?.AddBreadcrumb(
                    "BeforeSend callback failed.",
                    category: "SentryClient",
                    data: data,
                    level: BreadcrumbLevel.Error);
```